samples/config_helper_vulkan.cc

```diff
@@ -662,7 +669,20 @@ amber::Result ConfigHelperVulkan::ChooseVulkanPhysicalDevice(
   }
 
   for (uint32_t i = 0; i < count; ++i) {
-    vkGetPhysicalDeviceFeatures(physical_devices[i], &available_features_);
+    if (use_physical_device_features2_) {
+      VkPhysicalDeviceFeatures2KHR features2 = {};
+      features2.sType = VK_STRUCTURE_TYPE_PHYSICAL_DEVICE_FEATURES_2_KHR;
+      features2.pNext = nullptr;
+
+      auto vkGetPhysicalDeviceFeatures2KHR =
+          reinterpret_cast<PFN_vkGetPhysicalDeviceFeatures2KHR>(
+              vkGetInstanceProcAddr(vulkan_instance_,
+                                    "vkGetPhysicalDeviceFeatures2KHR"));
+      vkGetPhysicalDeviceFeatures2KHR(physical_devices[i], &features2);
```

Comment:
Will this code even link if physical-device-features2 is not supported?

Reply:
As long as the structures are in the header file it's all good. For the actual function call, we use the getInstanceProcAddr call, so that should be fine as long as the PFN_vkGetPhysicalDeviceFeatures2KHR  in the header as well.

I used the KHR version for everything to make sure it was backwards compatible.